src/main/java/triss/Triss.java

```diff
@@ -0,0 +1,429 @@
+        }
+
+        // Throw error if user did not type in a timing for the event
+        if (eventTiming.isBlank()) {
+            throw new TrissException("You didn't insert a date in your event! Let's try that again.");
+        }
+
+        // Add event to tasks
+        Event newEvent = new Event(taskName, eventTiming);
+        tasks.add(newEvent);
+
+        // Then, echo the task if not silent
+        if (!isSilent) {
+            printLine("I've added: " + newEvent.printTask());
+        }
+    }
+
+    /**
+     * Creates a new deadline based on the user's input.
+     * User has to type the input in this format: "deadline Meet with Friends /12th July".
+     * If the user types incorrectly, it asks the user to try again.
+     * @param userInput Any user input starting with the word "deadline".
+     */
+    private static void createNewDeadline(String userInput, boolean isSilent) throws TrissException {
+        String deadlineDate;
+        String taskName;
+
+        try {
+            deadlineDate = userInput.substring(userInput.indexOf("/") + 1).trim();
+            taskName = userInput.substring(END_INDEX_OF_WORD_DEADLINE, userInput.indexOf("/")).trim();
+        } catch (Exception error) {
+            String errorMessage = "You didn't write your deadline properly!\n"
+                    + " \n"
+                    + "Try inserting a deadline in this format:\n"
+                    + "    deadline Meet with Friends /12th July";
+            throw new TrissException(errorMessage);
+        }
+
+        // Catch other possible errors
+        // Throw error if user did not type in a name for the task
+        if (taskName.isBlank()) {
+            throw new TrissException("Your deadline name is blank! Let's try that again.");
+        }
+
+        // Throw error if user did not type in a timing for the event
+        if (deadlineDate.isBlank()) {
+            throw new TrissException("You didn't insert a date in your deadline! Let's try that again.");
+        }
+
+        // Add deadline to tasks
+        Deadline newDeadline = new Deadline(taskName, deadlineDate);
+        tasks.add(newDeadline);
+
+        // Then, echo the task if not silent
+        if (!isSilent) {
+            printLine("I've added: " + newDeadline.printTask());
+        }
+    }
+
+    /**
+     * Mark user task as done, if request is valid.
+     * Stops if user did not specify a task.
+     * Stops if user's chosen task does not exist.
+     * Informs user if task was already done.
+     * @param userInput Any user input starting with "done"
+     */
+    private static void handleUserMarkingTaskAsDone(String userInput) {
+        // Get number of task after the term "done"
+        int indexOfCompletedTask;
+        // Throw exception if user did not type a number after "done"
+        try {
+            indexOfCompletedTask = Integer.parseInt(parseUserInput(userInput, 1)) - 1;
+        } catch (Exception e) {
```

Comment:
if you have a try catch block at a parent method that calls this method, you could throw an error here so and catch the error at the parent method so that you do not need empty returns or so many try catch blocks. (Occurred many times in your code) Essentially, 1 try catch is possible for the entire code.

Reply:
Do you think it's okay if I have try catch blocks that throw errors? The benefit is that I don't need to do multiple if-else statements to check the validity of the input, and then throw errors.